test/functional/api/version_spec.lua

```diff
@@ -58,10 +58,18 @@ describe('api metadata', function()
     return by_name
   end
 
-  -- Remove metadata that is not essential to backwards-compatibility.
-  local function filter_function_metadata(f)
+  -- Remove or patch metadata that is not essential to backwards-compatibility.
+  local function normalize_func_metadata(f)
+    -- Dictionary was renamed to Dict. That doesn't break back-compat because clients don't actually
+    -- use the `return_type` field (evidence: "ArrayOf(…)" didn't break clients).
+    f.return_type = f.return_type:gsub('Dictionary', 'Dict')
+
     f.deprecated_since = nil
     for idx, _ in ipairs(f.parameters) do
+      -- Dictionary was renamed to Dict. Doesn't break back-compat because clients don't actually
+      -- use the `parameters` field of API metadata (evidence: "ArrayOf(…)" didn't break clients).
+      f.parameters[idx][1] = f.parameters[idx][1]:gsub('Dictionary', 'Dict')
```

Comment:
I suspect this won't break any clients to return the `return_type` in the `api_info` metadata. We already report `ArrayOf(..)` and similar stuff in `return_type`, and that didn't break clients.

I think clients are using the types from the actual RPC calls, not api-metadata. There might be some clients that are generating code from api-metadata, and this would be an easy fix for them (that's not really "breaking" if it's a build-time change).